pyro/infer/mcmc/hmc.py

```diff
@@ -75,16 +78,68 @@ def _energy(self, z, r):
 
     def _reset(self):
         self._t = 0
+        self._accept_cnt = 0
+        # TODO: move these parameters to self.defaults dict or self.config dict of init method
+        self._target_accept_prob = 0.8  # from Stan
+        self._adapted_step_size = self.step_size if self.step_size is not None else 1
+        self._trajectory_length = 2 * math.pi  # from Stan
+        self._t0 = 10
+        self._kappa = 0.75
```

Comment:
Can we remove these parameters as they are effectively constants (and we are not setting them via the constructor) that are replicated in `DualAveraging`? If for some reason, we need them to be different for HMC and NUTS, we can put them back in later.